### DISCUSSION TITLE
Creating a CDROM image

### DISCUSSION THREAD

**Discussion by ethindp** [phil-opp/blog_os#713] [Questions]
I need to build an Iso image of my kernel but I can't seem to find the article that talks about that. Could someone point me tothose instructions?

**Comment by phil-opp** [comment]
The `bootimage` tool creates a disk image that you can boot from an USB stick. To boot from a CD, you need to create an image that implements the "El-Torito" CD-booting standard. See the article in the OSDev wiki for more information: https://wiki.osdev.org/El-Torito

**Comment by ethindp** [comment]
Yeah, I figured that out, but I'm now suffering the part where the
bootloader gets stuck trying to boot at stage one. There's a related
issue in the bootloader repository for that one.

On 1/8/20, Philipp Oppermann <notifications@github.com> wrote:
> The `bootimage` tool creates a disk image that you can boot from an USB
> stick. To boot from a CD, you need to create an image that implements the
> "El-Torito" CD-booting standard. See the article in the OSDev wiki for more
> information: https://wiki.osdev.org/El-Torito
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/phil-opp/blog_os/issues/713#issuecomment-571956795


-- 
Signed,
Ethin D. Probst

**Comment by phil-opp** [comment]
You mean the issue about chainloading with GRUB?

**Comment by Lucky4Luuk** [comment]
Any update on this? I have the same problem

**Comment by phil-opp** [comment]
@Lucky4Luuk Could you clarify your problem? Are you trying to create a bootable CD image to burn it to a CD?

**Comment by Lucky4Luuk** [comment]
I am not having any specific problem, but I'd like more information on what he did to compile his code into an ISO (or possibly convert the bin file to an iso) and how he fixed the problem with it being stuck at stage one of the booting process. When I used qemu to turn my bin file into a vdi file for virtual box, I get the exact same problem with it being stuck at stage one of the boot process. Not necessarily something you can fix, I think, but just wondering what this guy did to fix his issues.

**Comment by phil-opp** [comment]
Thanks for clarifying!

**Comment by ethindp** [comment]
I didn't; I can't seem to figure it out myself. People have suggested moving it around on partitions and stuff like that but I'd (hopefully) rather avoid over-complicating the process, in case I want to automate it in the future.
